fastify.js

```diff
@@ -423,7 +436,14 @@ function build (options) {
       )
 
       try {
-        buildSchema(context, opts.schemaCompiler || _fastify._schemaCompiler)
+        buildSchema(
+          context,
+          opts.schemaCompiler || _fastify._schemaCompiler,
+          // using the getSchema to re-use the cache created for that
+          opts.schemaResolver || _fastify.getSchema,
```

Comment:
I think this may be related to #611 so the `opts.schemaResolver` part isn't necessary.

Reply:
Oh wait, I just saw the _"you can pass it in either on route options or through `setSchemaResolver`"_ note in the documentation. In that case, you'll need to add the `schemaResolver` option after [this line](https://github.com/fastify/fastify/blob/889530b38c36833b30d3a7db6c0a83c493e85c6c/fastify.js#L378), a test to make sure the option works, and also documentation for the option to the [Route docs](https://github.com/fastify/fastify/blob/master/docs/Routes.md#full-declaration).